lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -360,7 +376,7 @@ public function execute($params = null)
             $hasZeroIndex = array_key_exists(0, $params);
             foreach ($params as $key => $val) {
                 if ($hasZeroIndex && is_numeric($key)) {
-                    $this->bindValue($key + 1, $val);
+                    $this->bindValue((int) $key + 1, $val);
```

Comment:
Maybe we should document `$params` as `array<int, mixed>` instead?

Reply:
This should apply to any cast, by the way: casting leads to silent failures in way too many scenarios

Reply:
> This should apply to any cast, by the way: casting leads to silent failures in way too many scenarios

Hm… in this particular case, we're casting to an integer something which `is_numeric()`, and we're using it only because there's no `is_integerish()` in PHP. Pracically, I can't see any problem here.

Reply:
I'd surely get rid of the cast though - even asserting would be better.

Reply:
Actually, I'm even sure which assertion would help here. In fact, it looks like a false-positive on the PHPStan end (see the [example](https://phpstan.org/r/2edaabe7-40cb-4a94-a312-d6024d9a55b0)). Adding a number to a numeric string won't result in an error.

But I think we can replace `is_numeric()` with `is_int()`: since the value comes from an array key, the only numeric but non-integer value would be something like `'1.5'` which is not a valid positional or named parameter anyways.